extend-expect.d.ts

```diff
@@ -1,4 +1,24 @@
 declare namespace jest {
+  interface InverseStringAsymmetricMatchers {
+    stringMatching(str: string | RegExp): any
+    stringContaining(str: string): any
+  }
+  interface Expect {
+    stringMatching(str: string | RegExp): any
+    stringContaining(str: string): any
+    anything(): any
+    any(classType: any): any
+    not: InverseStringAsymmetricMatchers
+  }
+
+  type attributeValueType =
+    | string
+    | Expect.stringContaining
+    | Expect.stringMatching
+    | Expect.any
+    | Expect.anything
+    | Expect.not
```

Comment:
@gnapse I am not sure whether there is a better way yo do this?

The interface `Expect` is declared by looking into [@types/jest](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d8748201c7bfc38d70269af3bcaa3a5c380547af/types/jest/index.d.ts#L488)
But for at least vscode this types is shown as `any`.

Reply:
Hmmm, I would've expected that since this is a feature of `jest`, its type definitions would already give types for these constructs instead of us having to provide it. I'll check jest's type definitions later to look into this.

Reply:
Sorry. I haven't had time to check for this. I'm ok with publishing this for the time being, and we can always enhance it or change it to use more official types later.